core/src/avm2/globals/XML.as

```diff
@@ -118,9 +121,13 @@ package {
             return self.AS3::name();
         };
 
-        prototype.namespace = function(prefix:String = null):* {
+        prototype.namespace = function(...args):* {
```

Comment:
same here, you could keep the correct signature and pass the `arguments` via `apply()`.

Reply:
I can call `XML.namespace()` with any number of arguments in FP, so I'm not sure what you're stating is correct

Reply:
I don't know if we reproduce this correctly in Ruffle, but the presence of `arguments` prevents too-many-args error in FP.

```as
class C{ public static function f(a: String) { } }
var f = C.f;
f(); // error
f("a"); // ok
f("a", "b", "c"); // error

class C{ public static function f(a: String=null) {  } }
var f = C.f;
f(); // ok
f("a"); // ok
f("a", "b", "c"); // error

class C{ public static function f(a: String=null) { trace(arguments) } }
var f = C.f;
f(); // ok
f("a"); // ok
f("a", "b", "c"); // ok
```

Reply:
(also unrelated to this topic, but for completeness: a fully untyped function like `var f = function(a){}` also skips argument count checks. Adding a default `a=null` or type `a: String` reenables it. I don't know the exact rules here though.)

Reply:
So I think `arguments` is the only way we make it "variadic" while keeping `describeType`-compatible signature.

Reply:
https://github.com/ruffle-rs/ruffle/pull/13179#discussion_r1328458053 this sounds like a function verification thing

Reply:
> This sounds like a function verification thing, which we haven't implemented.

I think we have this implemented? Didn't test but we have this logic:
```
        let has_rest_or_args = method.is_variadic(); // <- true if uses `arguemnts`

        if user_arguments.len() > signature.len() && !has_rest_or_args {
            return Err(format!(
                "Attempted to call {:?} with {} arguments (more than {} is prohibited)",
```

Reply:
Whoops, my mistake. Apparently this exists: `AbcMethodFlags::NEED_ARGUMENTS`

Reply:
I don't think any of the correct signature or the number of arguments is really going to matter in practice. Considering that you have to do a bunch of tricks to even bypass the arguments length check of the compiler, I would just go for the correct signature?

Honestly I don't have any great insight here :/